--- /tmp/original-strings.lua	2026-09-13 10:02:30.087611840 -0700
+++ /tmp/refactored-strings.lua	2026-09-13 10:02:30.084611829 -0700
@@ -108,7 +108,7 @@
   end
 end)()
 
-local truncate = function(str, len, dots, direction)
+local truncate_directional = function(str, len, dots, direction)
   if M.strdisplaywidth(str) <= len then
     return str
   end
@@ -141,15 +141,15 @@
   dots = dots or "…"
   direction = direction or 1
   if direction ~= 0 then
-    return truncate(str, len, dots, direction)
+    return truncate_directional(str, len, dots, direction)
   else
     if M.strdisplaywidth(str) <= len then
       return str
     end
     local len1 = math.floor((len + M.strdisplaywidth(dots)) / 2)
-    local s1 = truncate(str, len1, dots, 1)
+    local s1 = truncate_directional(str, len1, dots, 1)
     local len2 = len - M.strdisplaywidth(s1) + M.strdisplaywidth(dots)
-    local s2 = truncate(str, len2, dots, -1)
+    local s2 = truncate_directional(str, len2, dots, -1)
     return s1 .. s2:sub(dots:len() + 1)
   end
 end
